crates/hir-def/src/body/lower.rs

```diff
@@ -607,6 +627,16 @@ impl ExprCollector<'_> {
         })
     }
 
+    fn initialize_binding_owner(
+        &mut self,
+        syntax_ptr: AstPtr<ast::Expr>,
+    ) -> (ExprId, Option<ExprId>) {
+        let result_expr_id = self.alloc_expr(Expr::Missing, syntax_ptr);
+        let prev_binding_owner = self.current_binding_owner.take();
+        self.current_binding_owner = Some(result_expr_id);
+        (result_expr_id, prev_binding_owner)
+    }
```

Comment:
I assume we'll fix this up properly once we do late name res similar to rust (where this is equivalent to a rib I think)?

Reply:
It will be fixed when we separate const bodies from their parent. If we want to fix it now, we can do it by making a `collect_expr_into` function which put the expr into the provided id in arena.

I don't see how it is related to name resolution?

Reply:
Right, nevermind I was thinking of name res as we would have to prevent resolution going across these